crates/bitwarden-uniffi/src/vault/mod.rs

```diff
@@ -37,4 +39,14 @@ impl ClientVault {
     pub fn sends(self: Arc<Self>) -> Arc<sends::ClientSends> {
         Arc::new(sends::ClientSends(self.0.clone()))
     }
+
+    /// Generate a TOTP code from a provided key.
+    ///
+    /// The key can be either:
+    /// - A base32 encoded string
+    /// - OTP Auth URI
+    /// - Steam URI
+    pub async fn generate_totp(&self, key: String) -> TotpResponse {
+        self.0 .0.read().await.vault().generate_totp(key).await
+    }
```

Comment:
I wasn't sure where to put this, it realistically only for vault ciphers but it's more similar to cryptography then anything else.

Reply:
I think it's fine in vault, we can even put it in the ciphers subclient inside vault too.